exatomic/gaussian/output.py

```diff
@@ -608,6 +619,212 @@ def parse_momatrix(self):
+        nframes = int(len(fx)/nat)
+        # generate frame labels
+        frame = [i for i in range(nframes)]
+        # extend to match sizes of all arrays
+        frame = np.repeat(frame, nat)
+        label = np.tile(label, nframes)
+        znums = np.tile(znums, nframes)
+        symbols = np.tile(symbols, nframes)
+        # create dataframe
+        self.gradient = pd.DataFrame.from_dict({"Z": znums, "label": label, "fx": fx, "fy": fy,
+                                                "fz": fz, "symbol": symbols, "frame": frame})
+
+    def parse_shielding_tensor(self):
+        # nmr shielding regex
+        _renmr = 'NMR shielding'
+        # base properties
+        _renat = 'Number of atoms'
+        _reznums = 'Atomic numbers'
+
+        found = self.find(_renmr)
+        if not found:
+            return
+        else:
+            found = self.find(_renmr, _renat, _reznums, keys_only=True)
+        # get number of atoms
+        nat = self._intme(found[_renat])
+        # get atomic numbers
+        znums = self._dfme(found[_reznums], nat)
+        # generate symbols
+        symbols = list(map(lambda x: z2sym[x], znums))
+        # generate labels
+        label = [i for i in range(len(znums))]
+        # get NMR shielding tensors (Hz)
+        # TODO: Check that this is in fact in Hz
+        shield = self._dfme(found[_renmr], nat * 9)
+        shield[abs(shield) < self.tol] = 0
+        # arrange into x, y, z cols
+        x = shield[::3]
+        y = shield[1::3]
+        z = shield[2::3]
+#        # conversion from Hz to ppm
+#        # (only done for some test calculations may not be accurate for all)
+#        x *= 1e6/18778.86
+#        y *= 1e6/18778.86
+#        z *= 1e6/18778.86
+        matrix = np.transpose([x,y,z]).reshape(nat,3,3)
+        matrix = np.asarray([0.5*(matrix[i] + np.transpose(matrix[i])) for i in range(len(matrix))])
+        # compute isotropic and anisotropic values
+        # done in units of Hz
+        iso = np.zeros(len(matrix))
+        aniso = np.zeros(len(matrix))
+        # TODO: check when we get complex eigenvalues
+        for i in range(len(matrix)):
+            vals = np.linalg.eigvals(matrix[i])
+#            vals = np.real(vals)
+            vals.sort()
+            iso[i] = np.average(vals, axis=-1)
+            aniso[i] = vals[2] - (vals[0] + vals[1])/2
+            # for debugging
+            #conv = 1e6/18778.86
+            #print("====================={}=====================".format(i+1))
+            #print("Eigenvalues:\t{}\t{}\t{}".format(vals[0]*conv, vals[1]*conv, vals[2]*conv))
+            #print("Isotropic:\t{}\nAnisotropy:\t{}".format(iso[i]*conv, aniso[i]*conv))
+        matrix = matrix.reshape(len(matrix)*3, -1)
+        df = pd.DataFrame(matrix, columns=['x', 'y', 'z'])
+        label = 'nmr shielding'
+        # get atom center indexes
+        atom = [i for i in range(nat)]
+        # get atom symbols
+        symbols = list(map(lambda x: z2sym[x], znums))
```

Comment:
Since this will be passed to `np.tile`, there is no need to greedy evaluate the generator here. I would suggest removing the `list` as `np.tile` (below) will evaluate the generator (i.e. use the lazy evaluation mentality that generators are good for!).

Reply:
Yeah, that makes more sense I think I was just trying to use some of the things that were already being done that I know worked and went with it.